internal/testcomposer/testcomposer.go

```diff
@@ -152,3 +167,47 @@ func (c *Client) doJSONResponse(req *http.Request, expectStatus int, v interface
 
 	return json.NewDecoder(res.Body).Decode(v)
 }
+
+// CheckFrameworkAvailability checks that the requested is available on the backend
+func (c *Client) CheckFrameworkAvailability(ctx context.Context, frameworkName string) error {
+	url := fmt.Sprintf("%s/v1/testcomposer/framework/%s", c.URL, frameworkName)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return err
+	}
+	req.SetBasicAuth(c.Credentials.Username, c.Credentials.AccessKey)
+	resp, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	err = c.checkFrameworkRestrictions(*resp, string(body))
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("unexpected response code:'%d', msg:'%s'", resp.StatusCode, string(body))
+	}
+	return nil
+}
+
+// checkFrameworkRestrictions checks specific cases related to framework availability
+func (c *Client) checkFrameworkRestrictions(resp http.Response, body string) error {
+	if resp.StatusCode == http.StatusForbidden && body == forbiddenPreviewError {
+		log.Error().Msg("Looks like you are not part of the preview. To join the preview, please sign up here: https://info.saucelabs.com/scale-cypress-testing.html")
```

Comment:
Maybe say what's a preview for. Something like `cypress preview` or `testrunner preview`.

Reply:
I would also include (if this is possible) the username in the log message. `User "johndoe" is not registered for the Cypress/TestRunner preview`

Reply:
Preview is check at framework level, so it will be only `cypress preview`.
I'll use use log suggestion that fits the needs !